src/expression/expression.cpp

```diff
@@ -132,7 +132,7 @@ std::optional<ExpressionValue> Expression::evaluate() const
     if (rootNode_)
         return rootNode_->evaluate();
 
-    return std::nullopt;
+    return ExpressionValue();
```

Comment:
For the purposes of being ultra-clear on what we're guarding against here, I'd perhaps suggest increasing the payload of this PR by a few hundred percent and writing (as the whole function):
```suggestion
    // If there is no rootNode_ (i.e. expression) this is valid, and we just return an empty value
    if (!rootNode_)
        return ExpressionValue();

    return rootNode_->evaluate();
```